File: binance4j-c2c/src/main/java/com/binance4j/c2c/dto/TradeHistory.java
package com.binance4j.c2c.dto;

import java.util.List;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;

/**
 * C2C trades history.
 * 
 * @param code    Code.
 * @param message Message.
 * @param data    Trades.
 * @param total   Total.
 * @param success Success.
 */
@ApiModel("C2C trades history.")
public record TradeHistory(@ApiModelProperty("Code.") String code, @ApiModelProperty("Message.") String message, @ApiModelProperty("Trades.") List<Trade> data,
		@ApiModelProperty("Total.") int total, @ApiModelProperty("Success.") boolean success) {
}

File: src/main/kotlin/com/binance4j/connectors/c2c/dto/TradeHistory.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package com.binance4j.connectors.c2c.dto

import com.fasterxml.jackson.annotation.JsonProperty

import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty

/**
 * C2C trades history.
 *
 * @property code Code.
 * @property message Message.
 * @property data Trades.
 * @property total Total.
 * @property success Success.
 */
@ApiModel("C2C trades history.")
data class TradeHistory(
	@ApiModelProperty("Code.")
	@JsonProperty("code")
	var code: String,
	@ApiModelProperty("Message.")
	@JsonProperty("message")
	var message: String,
	@ApiModelProperty("Trades.")
	@JsonProperty("data")
	var data: List<Trade>,
	@ApiModelProperty("Total.")
	@JsonProperty("total")
	var total: Int,
	@ApiModelProperty("Success.")
	@JsonProperty("success")
	var success: Boolean
)